FlowCal/mef.py

```diff
@@ -265,26 +319,30 @@ def fit_beads_autofluorescence(fl_channel, fl_mef):
     # Initial guesses:
     # 0: slope found by putting a line through the highest two points.
     # 1: y-intercept found by putting a line through highest two points.
-    # 2: bead autofluorescence initialized to 100.
+    # 2: bead autofluorescence initialized using the first point.
     params[0] = (np.log(fl_mef[-1]) - np.log(fl_mef[-2])) / \
-                    (fl_channel[-1] - fl_channel[-2])
-    params[1] = np.log(fl_mef[-1]) - params[0] * fl_channel[-1]
-    params[2] = 100.
+                (np.log(fl_au[-1]) - np.log(fl_au[-2]))
+    params[1] = np.log(fl_mef[-1]) - params[0] * np.log(fl_au[-1])
+    params[2] = np.exp(params[0]*np.log(fl_au[0]) + params[1]) - fl_mef[0]
 
     # Error function
     def err_fun(p, x, y):
-        return np.sum((np.log(y + p[2]) - ( p[0] * x + p[1] ))**2)
-        
+        return np.sum((np.log(y + p[2]) - ( p[0] * np.log(x) + p[1] ))**2)
+
     # Bead model function
     def fit_fun(p,x):
-        return np.exp(p[0] * x + p[1]) - p[2]
+        return np.exp(p[0] * np.log(x) + p[1]) - p[2]
 
-    # Channel-to-MEF standard curve transformation function
+    # a.u.-to-MEF standard curve transformation function
+    # We use the equivalent form ``exp(p[1]) * x**p[0]``. Mathematically, this
+    # is undefined for x < 0. In this case, we redefine the output as
+    # ``- exp(p[1]) * (-x)**p[0]``. These two can be combined as follows:
+    # ``np.sign(x) * np.exp(p[1]) * (np.abs(x)**p[0])``.
     def sc_fun(p,x):
-        return np.exp(p[0] * x + p[1])
+        return np.sign(x) * np.exp(p[1]) * (np.abs(x)**p[0])
```

Comment:
Thought about this some more, and it actually kinda points out the usefulness of a logicle plot for the standard curve.

At the end of the day, we are extending the standard curve beyond 0; it would be nice to visualize that extension (i.e. plot the full standard curve, sufficiently extended into the negative RFI values to get an idea for how it behaves).

* one option would be to reflect the standard curve's behavior across the x-axis.
* perhaps equivalently, we could embrace the same underlying rationale for the logicle plot: linearize our function around 0 and extend it into the negative numbers.
 * is that the same as what you've done by assuming `p0 ~= 1` => `fl_mef = - exp(p1)*[abs(fl_rfi)**p0]`? I guess not technically, since you still have the `**p0` term. Does this introduce a discontinuitinty in the derivative of this function / make it not smooth? Keeping the `**p0` term seems like it will either introduce a discontinuity or mitigate the introduction of a discontinuity, but i'm not sure which. Would be nice to be smooth around 0, though (although I don't have an immediate application which requires that).
 * linearized version might just be: `fl_mef = - exp(p1) * abs(fl_rfi)`?
* could also include some more derivatives when you interpolate past 0? i.e. Taylor (Maclourin?) expansion? - I don't think this would be terribly useful, though
* I feel like reflecting/linearizing/assuming `p0~=1` is more appropriate than only using the real part. Invoking this complex logarithm doesn't feel appropriate for our situation; feels like a weird hack tied to this strange `p0` term in our model, not something fundamentally represented in our underlying data.

Practically speaking, though, it would help to see how these solutions look alongside each other (which again probably requires logicle plot of standard curve? can't see 0 or negative behavior on log plot. or maybe just linear plot that is zoomed in)